### PR TITLE
Allow custom types

### PR DESCRIPTION
**Pull request by rsmylskibc**
- Updated parseSchema() to allow using custom types
- Updated joi to latest (13.1.0) to allow for using extend()
- Added test for custom using type

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -117,14 +117,23 @@ function parseSchema( value, engine ) {
...
 
-    let parser = types[ type ];
+    let schemaParser = null;
 
-    if( !parser ) {
+    // Predefined type.
+    if (types[type]) {
+        schemaParser = types[type];
+    }
+    // Custom type - use Base schema parser.
```
Comment (rsmylskibc) [inline lib/parser.js:126]: I will update the comment here to better define what the check for `utils.isFunction(engine[type])` does/signifies.

Ultimately, types are just functions on the base engine (joi). The original lib has some special parsing for some joi types, etc, but fore the most part, the type and its options are pretty much all functions, and that's what the parser does.

For adding a custom type, by checking that the type exists as a function, is good enough to use it with no additional functionality changes, and let the parser process the config as per any other normal/basic type.

File: test/lib/index.test.js [hunk 2]
```diff
@@ -277,4 +277,47 @@ describe( 'lib/index', function() {
...
+
+        beforeEach(function() {
+            parser = index.parser(customJoi);
+        });
+
+        it("custom type valid and required", function() {
+            const schema = parser.parse({
+                "@type": "string20",
+                // use null to call with no params.
+                valid: null,
```
Comment (gburtini) [inline test/lib/index.test.js:312]: This feels unintuitive. Was it original functionality that you're just replicating?

Comment (rsmylskibc) [reply test/lib/index.test.js:312]: Which part being unintiutive - using null to tell the system to call the function without an argument?

If so, that's part of the original functionality. `lib/base.js updateSchema()`

Comment (gburtini) [reply test/lib/index.test.js:312]: Yes, sounds good then. :+1:

**Comment by gburtini** [comment]
LGTM. :shipit:

**Pull request merged**